packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -146,414 +150,242 @@ impl PostgresDatabase {
         Ok(())
     }
 
-    /// Inserts a [`DataType`] identified by [`VersionId`], and associated with an [`AccountId`],
-    /// into the database.
+    /// Inserts the specified [`DatabaseType`].
+    ///
+    /// This first checks if the [`BaseUri`] of the type does not already exist. If not, it will
+    /// insert the [`BaseUri`] and creates a new [`VersionId`] from the contained [`VersionedUri`]
+    /// and inserts the type.
     ///
     /// # Errors
     ///
-    /// - [`DatastoreError`], if inserting failed.
-    async fn insert_data_type(
-        &mut self,
-        version_id: VersionId,
-        data_type: &DataType,
+    /// - If the [`BaseUri`] already exists
+    ///
+    /// [`BaseUri`]: crate::types::BaseUri
+    async fn create<T>(
+        &self,
+        database_type: T,
```

Comment:
If we do change the name of `type` to something else, we should consider renaming the `database_type` properties in various locations to that new name. Putting a comment for persistance. Related to this thread https://github.com/hashintel/hash/pull/805/files#r921054810